source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -126,113 +130,93 @@ used at the data preparation and data processing stages for the following:
 - To transfer memory ownership of the data from user application to the table,
   or share it between them.
 
-- To connect with the :txtref:`data-source` to convert from an out-of-memory
-  into an in-memory dataset representation.
+- To connect with the :txtref:`data-source` to convert data from an
+  out-of-memory into an in-memory representation.
 
 - To support streaming of the data to the algorithm.
 
 - To access the underlying data on a device in a required :capterm:`data
   format`, e.g. by blocks with the defined :capterm:`data layout`.
 
-For thread-safety reasons and better integration with external entities, a table
-provides a read-only access to the data within it, thus, table concept
-implementations shall be :capterm:`immutable <immutability>`.
+.. note::
+  For thread-safety reasons and better integration with external entities, a table
+  provides a read-only access to the data within it, thus, table concept
+  implementations shall be :capterm:`immutable <immutability>`.
+
+.. warning::
+  Table can represent not a datasets only - it can be a matrix, a vector (if
```

Comment:
do we want to say that the table represents not only the data, but the result of the data processing in the form of matrix, vector etc?